pandas_gbq/gbq.py

```diff
@@ -837,68 +681,172 @@ def _get_credentials_file():
         'PANDAS_GBQ_CREDENTIALS_FILE')
 
 
-def _parse_data(schema, rows):
-    # see:
-    # http://pandas.pydata.org/pandas-docs/dev/missing_data.html
-    # #missing-data-casting-rules-and-indexing
-    dtype_map = {'FLOAT': np.dtype(float),
-                 'TIMESTAMP': 'M8[ns]'}
+def sizeof_fmt(num, suffix='B'):
+    fmt = "%3.1f %s%s"
+    for unit in ['', 'K', 'M', 'G', 'T', 'P', 'E', 'Z']:
+        if abs(num) < 1024.0:
+            return fmt % (num, unit, suffix)
+        num /= 1024.0
+    return fmt % (num, 'Y', suffix)
 
-    fields = schema['fields']
-    col_types = [field['type'] for field in fields]
-    col_names = [str(field['name']) for field in fields]
-    col_dtypes = [dtype_map.get(field['type'], object) for field in fields]
-    page_array = np.zeros((len(rows),), dtype=lzip(col_names, col_dtypes))
-    for row_num, raw_row in enumerate(rows):
-        entries = raw_row.get('f', [])
-        for col_num, field_type in enumerate(col_types):
-            field_value = _parse_entry(entries[col_num].get('v', ''),
-                                       field_type)
-            page_array[row_num][col_num] = field_value
 
-    return DataFrame(page_array, columns=col_names)
+def run_query(query, client, dialect='legacy', query_parameters=(),
```

Comment:
Nit: this PR would be easier to review if the `run_query()` method didn't move (stayed defined just before `load_data()`)

Reply:
Good idea, moved this back to GbqConnector and moved away some more of the plumbing in read_gbq to GbqConnector as well.